linker/linker_test.go

```diff
@@ -2225,23 +2210,26 @@ func TestCustomJSONNameWarnings(t *testing.T) {
 
 // testByProtoc tests the proto files with protoc. The fileNames parameter indicates the order of file
 // that should be used in the command line for protoc. fileNames can be nil when the order does not matter.
-func testByProtoc(t *testing.T, files map[string]string) (passed bool, err error) {
+func testByProtoc(t *testing.T, files map[string]string, fileNames []string) (passed bool, err error) {
+	if len(fileNames) != 0 {
+		require.True(t, len(files) == len(fileNames))
+		for _, fileName := range fileNames {
+			_, ok := files[fileName]
+			require.True(t, ok)
+		}
+	}
+
 	tempDir := writeFileToDisk(t, files)
 	defer os.RemoveAll(tempDir)
-
-	fileNames := make([]string, 0, len(files))
+	if len(fileNames) == 0 {
+		fileNames = make([]string, 0, len(files))
+	}
 	for fileName := range files {
 		fileNames = append(fileNames, fileName)
 	}
```

Comment:
I think this loop is supposedly be inside the `if len(fileNames) == 0 {` block above.

Reply:
ah yes this is messing up the order